### PR TITLE
feat: ugov module proto

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

New `ugov` module to manage umee / chain general parameters. 
Notes:
* I'm still not sure about the best name for the module. Please share your opinion here. Other options I was thinking about:  `x/chaingov` or `x/ext` or `x/umee` or `x/umeegov` or `x/ugov`.
* This module probably will be also used for emergency groups etc... 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/umee/ugov/v1/query.proto [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+option go_package = "github.com/umee-network/umee/v4/x/ugov";
+
+option (gogoproto.goproto_getters_all) = false;
+
+// Query defines the gRPC querier service.
+service Query {
+  // MinTxFees returns minimum transaction fees.
+  rpc MinTxFees(QueryMinTxFees) returns (QueryMinTxFeesResponse) {
+    option (google.api.http).get = "/umee/ugov/v1/params";
```
Comment (toteki) [inline proto/umee/ugov/v1/query.proto:17]: url needs update

**Review by gsk967 — CHANGES_REQUESTED** [review]
May be we should call `ugen` (Umee General Params) instead of ugov

File: proto/umee/ugov/v1/tx.proto [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+option go_package = "github.com/umee-network/umee/v4/x/ugov";
+
+option (gogoproto.goproto_getters_all) = false;
+// option (gogoproto.messagename_all)     = true;
+
+// Msg defines the x/ugov module's Msg service.
+service Msg {
+  // GovUpdateMinFees sets protocol controlled tx min fees.
+  rpc GovMinFees(MsgGovUpdateMinFees) returns (MsgGovUpdateMinFeesResponse);
```
Comment (toteki) [inline proto/umee/ugov/v1/tx.proto:17]:
```suggestion
  // GovUpdateMinFees sets protocol controlled tx min fees.
  rpc GovUpdateMinFees(MsgGovUpdateMinFees) returns (MsgGovUpdateMinFeesResponse);
```

File: proto/umee/ugov/v1/tx.proto [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import "cosmos/base/v1beta1/coin.proto";
+import "cosmos/msg/v1/msg.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/umee-network/umee/v4/x/ugov";
+
+option (gogoproto.goproto_getters_all) = false;
+// option (gogoproto.messagename_all)     = true;
```
Comment (toteki) [inline proto/umee/ugov/v1/tx.proto:12]:
```suggestion
```

**Comment by toteki** [comment]
Left some comments

`ugov` seems like the best option out of the ones I've seen so far

File: proto/umee/ugov/v1/query.proto (hunk 1 above)
Comment (robert-zaremba) [reply proto/umee/ugov/v1/query.proto:17]: updated ulr to `min-tx-fees`.

**Review by gsk967 — APPROVED** [review]

**Review by toteki — APPROVED** [review]

**Review by toteki — APPROVED** [review]

**Pull request merged**